src/tx/write_tx.rs

```diff
@@ -689,15 +730,42 @@ impl<'a> WriteTransaction<'a> {
                 // that way, the write is definitely always the newest
                 SeqNo::MAX,
             ));
+        self.conflict_manager
+            .mark_conflict(&partition.inner.name, &key.as_ref().into());
     }
 
     /// Commits the transaction.
     ///
     /// # Errors
     ///
     /// Will return `Err` if an IO error occurs.
-    pub fn commit(self) -> crate::Result<()> {
-        let mut batch = Batch::new(self.keyspace).durability(self.durability);
+    pub fn commit(mut self) -> crate::Result<()> {
```

Comment:
I think it would be nice for `commit` to return a `Result<Result<()>>`, the outer one being IO errors, the inner one being the transaction conflict error, that way you don't need to match on the error and instead can do:

```rs
loop {
  let tx = ...
  ...
  if tx.commit()?.is_ok() {
      break;
  }
}
```
If a transient I/O error actually happens, just continuing the loop would be pretty pointless because the journal is likely poisoned from that point on.